mapit_gb/management/commands/mapit_UK_show_missing_codes.py

```diff
@@ -30,18 +46,25 @@ def handle_noargs(self, **options):
                 type=area_type
             )
             for code_type in used_code_types:
-                areas_without_codes = areas.exclude(codes__type__code=code_type)
-                if areas_without_codes.count() > 0:
-                    print '  {count} {type} areas have no {code_type} code:'.format(
-                        count=areas_without_codes.count(),
-                        type=area_type,
-                        code_type=code_type
-                    )
-                    for area in areas_without_codes:
-                        print '    {name} {id} (gen {gen_low}-{gen_high}) {country}'.format(
-                            name=area.name,
-                            id=area.id,
-                            gen_low=area.generation_low_id,
-                            gen_high=area.generation_high_id,
-                            country=area.country
+                if self.is_code_type_required_for_area_type(code_type, area_type):
+                    areas_without_codes = areas.exclude(codes__type__code=code_type)
+                    if areas_without_codes.count() > 0:
+                        print '  {count} {type} areas have no {code_type} code:'.format(
+                            count=areas_without_codes.count(),
+                            type=area_type,
+                            code_type=code_type
                         )
+                        for area in areas_without_codes:
+                            print '    {name} {id} (gen {gen_low}-{gen_high}) {country}'.format(
+                                name=area.name,
+                                id=area.id,
+                                gen_low=area.generation_low_id,
+                                gen_high=area.generation_high_id,
+                                country=area.country
+                            )
+
+    def is_code_type_required_for_area_type(self, code_type, area_type):
+        if code_type == "govuk_slug" and area_type == "EUR":
```

Comment:
I would simplify this to:

```python
return not (code_type == "govuk_slug" and area_type == "EUR")
```

Or you could reverse the logic here to have fewer negatives and describe more accurately what this method defines, so that it becomes:

```python
def is_code_type_not_required_for_area_type(self, code_type, area_type):
    return (code_type == "govuk_slug" and area_type == "EUR")
```

...and then we'd call it like this from the loop above:

```python
continue if self.is_code_type_not_required_for_area_type(code_type, area_type)
```